### PR TITLE
Add "numeric" type report filter to UCR

### PR DESCRIPTION
**Pull request by NoahCarnahan**
If you add this to your reports filters:

```
{
   "type": "numeric",
   "slug": "number_of_children_slug",
   "field": "number_of_children",
   "display": "Number of Children"
 }
```

You get this:
numeric filter

FYI @nickpell , the L1 report builder calls for this functionality.

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/reports_core/filters.py [hunk 1]
```diff
@@ -137,6 +137,44 @@ def filter_context(self):
         }
 
 
+class NumericFilter(BaseFilter):
+    template = "reports_core/filters/numeric_filter.html"
+
+    def __init__(self, name, required=True, label='Numeric Filter', css_id=None):
```
Comment (czue) [inline corehq/apps/reports_core/filters.py]: this should be marked for translation (either with `ugettext_lazy` or `ugettext_noop`)

File: corehq/apps/userreports/reports/filters.py [hunk 2]
```diff
@@ -39,6 +39,26 @@ def to_sql_values(self):
...
+
+    def __init__(self, filter, value):
+        assert filter.type == "numeric"
+        assert (isinstance(value, dict) and "operator" in value and "operand" in value) or value is None
+        super(NumericFilterValue, self).__init__(filter, value)
+
+    def to_sql_filter(self):
+        if self.value is None:
+            return ""
+        return "{0} {1} :operand".format(self.filter.field, self.value['operator'])
```
Comment (czue) [inline corehq/apps/userreports/reports/filters.py]: I wonder if this class should also have an assert on the operator. could potentially be open to sql-injection if someone misuses this.

**Comment by czue** [comment]
this looks great - just a couple small notes

File: corehq/apps/userreports/reports/filters.py (hunk 2 above)
Comment (NoahCarnahan) [inline corehq/apps/userreports/reports/filters.py]: I check the operator a little up stream in the `NumericFilter` class on [line 167](https://github.com/dimagi/commcare-hq/pull/5407/files#diff-e4b58799e85ecb0445e2ab040416a01eR167), which I think will prevent sql injections. Might make more sense to move that check into this class though...

Comment (czue) [inline corehq/apps/userreports/reports/filters.py]: yeah i saw that, but i think if someone else uses this class it'd be good for the assert to be here

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/userreports/reports/filters.py (hunk 2 above)
Comment (NoahCarnahan) [inline corehq/apps/userreports/reports/filters.py]: Good point. Do you think I should also remove the assert from `NumericFilter`, or do it twice?

File: corehq/apps/reports_core/filters.py (hunk 1 above)
Comment (NoahCarnahan) [inline corehq/apps/reports_core/filters.py]: (updated)

File: corehq/apps/userreports/reports/filters.py (hunk 2 above)
Comment (czue) [inline corehq/apps/userreports/reports/filters.py]: no strong feelings - i think both is fine

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by NoahCarnahan** [comment]
#5422

**Pull request merged**